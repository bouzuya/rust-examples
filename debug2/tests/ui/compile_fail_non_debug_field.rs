struct NonDebug;

#[derive(Debug)]
struct MyDebug {
    field: NonDebug,
}

fn main() {}
