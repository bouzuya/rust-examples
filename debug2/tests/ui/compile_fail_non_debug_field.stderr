error[E0277]: `NonDebug` doesn't implement `Debug`
 --> tests/ui/compile_fail_non_debug_field.rs:5:5
  |
3 | #[derive(Debug)]
  |          ----- in this derive macro expansion
4 | struct MyDebug {
5 |     field: NonDebug,
  |     ^^^^^^^^^^^^^^^ `NonDebug` cannot be formatted using `{:?}`
  |
  = help: the trait `Debug` is not implemented for `NonDebug`
  = note: add `#[derive(Debug)]` to `NonDebug` or manually `impl Debug for NonDebug`
help: consider annotating `NonDebug` with `#[derive(Debug)]`
  |
1 + #[derive(Debug)]
2 | struct NonDebug;
  |
